runtime/shared_common/shrinit.cpp

```diff
@@ -4849,4 +4857,42 @@ getModule(J9VMThread* vmThread, U_8* className, UDATA classNameLength, J9ClassLo
 	return module;
 }
 
+
+/**
+ * This function returns if the system is running low on free disk space
+ * @param[in] vm The current J9JavaVM
+ * @param[out] maxsize The maximum shared cache size allowed if the free disk space is low.
+ * 
+ * @return False if the space free disk space >= SHRINIT_LOW_DISK_SIZE. True otherwise.
+ */
+static bool
+isFreeDiskSpaceLow(J9JavaVM *vm, U_64* maxsize)
+{
+	char cacheDirName[J9SH_MAXPATH];
+	memset(cacheDirName, 0, J9SH_MAXPATH);
+	bool ret = true;
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	if (-1 == SH_OSCache::getCacheDir(PORTLIB, vm->sharedCacheAPI->ctrlDirName, cacheDirName, J9SH_MAXPATH, J9PORT_SHR_CACHE_TYPE_PERSISTENT)) {
+		goto done;
+	}
+	/* check for free disk space */
+	J9FileStatFilesystem fileStatFilesystem;
+	if (0 == j9file_stat_filesystem(cacheDirName, 0, &fileStatFilesystem)) {
+		if (fileStatFilesystem.freeSizeBytes >= SHRINIT_LOW_FREE_DISK_SIZE) {
+			ret = false;
+		} else {
+			Trc_SHR_INIT_isFreeDiskSpaceLow_DiskSpaceLow(fileStatFilesystem.freeSizeBytes);
+		}
+	} else {
+		Trc_SHR_INIT_isFreeDiskSpaceLow_StatFileSystemFailed(cacheDirName);
```

Comment:
the tracepoint should include error information, i.e. get_last_error, from j9file_stat_filesystem

Reply:
Fixed